workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationService.test.ts

```diff
@@ -357,4 +357,56 @@ describe('DynamicAuthorizationService', () => {
       ).rejects.toThrow(GroupNotFoundError);
     });
   });
+
+  describe('getIdentityPermissionsByIdentity', () => {
+    let sampleGroupId: string;
+    let sampleGroupType: IdentityType;
+
+    let sampleAction: Action;
+    let sampleEffect: Effect;
+    let sampleSubjectType: string;
+    let sampleSubjectId: string;
+    let sampleConditions: Record<string, JSONValue>;
+    let sampleFields: string[];
+    let sampleDescription: string;
+    let mockIdentityPermission: IdentityPermission;
```

Comment:
a lot of these look like duplicates of `createIdentityPermissions`. Can we extract them to a higher level block to reduce duplication?